content/docs/examples/advanced-egress/egress-gateway-mtls-origination/index.md

```diff
@@ -0,0 +1,533 @@
+    $ kubectl create -n istio-system secret generic nginx-ca-certs --from-file=my-nginx.mesh-external.svc.cluster.local/2_intermediate/certs/ca-chain.cert.pem
+    {{< /text >}}
+
+1.  Generate the `istio-egressgateway` deployment with a volume to be mounted from the new secrets. Use the same options
+    you used for generating your `istio.yaml`:
+
+    {{< text bash >}}
+    $ helm template install/kubernetes/helm/istio/ --name istio-egressgateway --namespace istio-system -x charts/gateways/templates/deployment.yaml --set gateways.istio-ingressgateway.enabled=false \
+    --set gateways.istio-egressgateway.secretVolumes[0].name=egressgateway-certs \
+    --set gateways.istio-egressgateway.secretVolumes[0].secretName=istio-egressgateway-certs \
+    --set gateways.istio-egressgateway.secretVolumes[0].mountPath=/etc/istio/egressgateway-certs \
+    --set gateways.istio-egressgateway.secretVolumes[1].name=egressgateway-ca-certs \
+    --set gateways.istio-egressgateway.secretVolumes[1].secretName=istio-egressgateway-ca-certs \
+    --set gateways.istio-egressgateway.secretVolumes[1].mountPath=/etc/istio/egressgateway-ca-certs \
+    --set gateways.istio-egressgateway.secretVolumes[2].name=nginx-client-certs \
+    --set gateways.istio-egressgateway.secretVolumes[2].secretName=nginx-client-certs \
+    --set gateways.istio-egressgateway.secretVolumes[2].mountPath=/etc/nginx-client-certs \
+    --set gateways.istio-egressgateway.secretVolumes[3].name=nginx-ca-certs \
+    --set gateways.istio-egressgateway.secretVolumes[3].secretName=nginx-ca-certs \
+    --set gateways.istio-egressgateway.secretVolumes[3].mountPath=/etc/nginx-ca-certs > \
+    ./istio-egressgateway.yaml
+    {{< /text >}}
+
+1.  Redeploy `istio-egressgateway`:
+
+    {{< text bash >}}
+    $ kubectl apply -f ./istio-egressgateway.yaml
+    deployment "istio-egressgateway" configured
+    {{< /text >}}
+
+1.  Verify that the key and the certificate are successfully loaded in the `istio-egressgateway` pod:
+
+    {{< text bash >}}
+    $ kubectl exec -it -n istio-system $(kubectl -n istio-system get pods -l istio=egressgateway -o jsonpath='{.items[0].metadata.name}') -- ls -al /etc/nginx-client-certs /etc/nginx-ca-certs
+    {{< /text >}}
+
+    `tls.crt` and `tls.key` should exist in `/etc/istio/nginx-client-certs`, while `ca-chain.cert.pem` in
+    `/etc/istio/nginx-ca-certs`.
+
+## Mutual TLS origination for egress traffic
+
+1.  Define a `ServiceEntry` for `my-nginx.mesh-external.svc.cluster.local`:
+
+    {{< text bash >}}
+    $ cat <<EOF | kubectl apply -f -
+    apiVersion: networking.istio.io/v1alpha3
+    kind: ServiceEntry
+    metadata:
+      name: nginx
+    spec:
+      hosts:
+      - my-nginx.mesh-external.svc.cluster.local
+      ports:
+      - number: 80
+        name: http-port
+        protocol: HTTP
+      - number: 443
+        name: http-port-for-tls-origination
+        protocol: HTTP
```

Comment:
See my comment in the HTTS blog.. we should have this as HTTPS. and ask applications to initiate connections to port 80, use a virtual service to forward from 80 to 443 and destination rule to initiate mtls to 443. Asking applications to use http://google:443 si very clunky. Lots of SDKs don't have such provisions. They typically allow http/https.

Reply:
Also, we don't need this service entry since the service already exists. In fact, you will have duplicate services because we don't allow service entries to overwrite what was objtained from the service registry

Reply:
@rshriram 

> Asking applications to use http://google:443 si very clunky

Agreed, I do not ask it here. This application uses port 80, port 443 in this service entry is probably not needed. 

Two issues:
1.  The service entry for the port 80 is required since the original service declares port 443 only. I guess the service entry is ignored for the port 443 where there is a duplication with the Kubernetes service.
2.  The idea of this example is to simulate mTLS to some external service. In that case, the ServiceEntry will be required for both ports. So I wanted to use the correct configuration for an external service, just to demonstrate it with a Kubernetes service.

Reply:
@rshriram I have fixed the port 443 to have HTTPS protocol and `https` name. I still propose to use the ServiceEntry, so this configuration will work for the cases when the external service is outside the cluster. Here I used an in-cluster out-of-mesh service just for simplicity of demonstration. A more complex demonstration would be to deploy NGINX in some other environment, in some other cluster, for example, but that would complicate the example.